docs/docs/typelevel.md

```diff
@@ -232,6 +229,35 @@ The following rewrite rules are performed when simplifying inlined bodies:
    Dropping a binding might make other bindings redundant. Garbage collection proceeds until no further bindings
    can be dropped.
 
+## Restrictions for Transparent and Typelevel Functions
+
+Transparent methods are effectively final; they may not be overwritten.
+
+If a transparent
+method has a toplevel match expression or a toplevel splice `~` on its right-hand side,
+it is classified as a typelevel method that can _only_ be executed at compile time. For typelevel methods two more restrictions apply:
+
+ 1. They must be always fully applied.
+ 2. They may override other methods only if one of the overridden methods is concrete.
+
+The right hand side of a typelevel method is never invoked by dynamic dispatch. As an example consider a situation like the following:
+```scala
+class Iterable[T] {
+  def foreach(f: T => Unit): Unit = ...
+}
+class List[T] extends Iterable[T] {
+  override transparent def foreach(f: T => Unit): Unit = ...
```

Comment:
Putting the keyword override here is misleading given the semantics described below. I would vote for calling this "shadow transparent" to clearly distinguish this from a regular override.

Reply:
I would be cleaner, I agree. But it means a new modifier.